be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -230,11 +230,13 @@ void ScanOperator::_detach_chunk_sources() {
 
 void ScanOperator::update_exec_stats(RuntimeState* state) {
     auto ctx = state->query_ctx();
-    ctx->update_pull_rows_stats(_plan_node_id, _pull_row_num_counter->value());
-    if (ctx != nullptr && _bloom_filter_eval_context.join_runtime_filter_input_counter != nullptr) {
-        int64_t input_rows = _bloom_filter_eval_context.join_runtime_filter_input_counter->value();
-        int64_t output_rows = _bloom_filter_eval_context.join_runtime_filter_output_counter->value();
-        ctx->update_rf_filter_stats(_plan_node_id, input_rows - output_rows);
+    if (ctx != nullptr) {
```

Comment:
In which case the query context is null?

Reply:
Feels like a defensive code.